docs/site/Parsing-requests.md

```diff
@@ -0,0 +1,202 @@
+}
+```
+
+### Validation
+
+Validations are applied on the parameters and the request body data. They also
+use OpenAPI specification as the reference to infer the validation rules.
+
+#### Parameters
+
+We have the data type safety check for the parameters parsed from header, path,
+and query. For example, if a parameter should be an integer, then a number with
+decimal like "1.23" would be rejected.
+
+You can specify a parameter's type by calling shortcut decorators of `@param`
+like `@param.query.integer()`. A list of available shortcuts can be found in the
+[api documents](https://apidocs.strongloop.com/@loopback%2fdocs/openapi-v3.html#param).
+Check out the section on
+[parameter decorators](Decorators.md#parameter-decorator) for instructions on
+how to decorate the controller parameter.
+
+Here are our default validation rules for each type:
+
+- number: validated by `isNaN(Number(data))`.
+- integer: validated by `Number.isInteger(data)`.
+- long: validated by `Number.isSafeInteger(data)`.
+- date-time: should be a valid date-time defined in
+  [RFC3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14).
+- date: should be a valid full-date defined in
+  [RFC3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14).
+- boolean: after converted to all upper case, should be one of the following
+  values: `TRUE`, `1`, `FALSE` or `0`.
+
+#### Request Body
+
+The data from request body is validated against its OpenAPI schema
+specification. We use module [AJV](https://github.com/epoberezkin/ajv) to
+perform the validation, which validates data with a JSON schema generated from
+the OpenAPI schema specification.
+
+Take again the operation replaceTodo for instance:
+
+```ts
+import {Todo} from './models';
+
+...
+  @put('/todos/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todo: Todo,
+  ): Promise<boolean> {
+    return await this.todoRepo.replaceById(id, todo);
+  }
+...
+```
+
+The request body specification is defined by applying `@requestBody()` to
+argument `todo`, and the schema specification inside it is inferred from its
+type `Todo`. The type is exported from a `Todo` model.
+
+{% include note.html title="Model documentation" content="See [model](Model.md) to know more details about how to decorate a model class" %}
+
+When the `PUT` method `/todo/{id}` gets called, the `todo` instance from the
+request body will be validated with a well defined specification.
+
+Section on [RequestBody Decorator](Decorators.md#requestbody-decorator) has a
+very detailed explanation of the various ways to provide a request body
+specification. Make sure you read that section to understand how to use
+`@requestBody()`.
+
+A few tips worth mentioning:
+
+- If a model property's type refers to another model, make sure it is also
+  decorated with `@model` decorator.
+
+- By the API first approach, you can also provide the request body specification
```

Comment:
nitpick: `By the API first approach` -> `If you're using API first development approach`